app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers.php

```diff
@@ -405,13 +236,17 @@ public function updateSubscriber($subscriber, $updateStatus = false)
                 $subscriber->setData("mailchimp_sync_modified", 0);
                 $saveSubscriber = true;
             } catch (MailChimp_Error $e) {
-                if ($newStatus === 'subscribed' && $subscriber->getIsStatusChanged() && !$helper->isSubscriptionConfirmationEnabled($storeId)) {
+                if ($newStatus === 'subscribed' && $subscriber->getIsStatusChanged()
```

Comment:
Use constant to validate "subscrbied" status.
Also extract condition to function "isSubscribed" or similar.